src/hotspot/share/opto/convertnode.cpp

```diff
@@ -219,6 +234,23 @@ Node *ConvF2LNode::Ideal(PhaseGVN *phase, bool can_reshape) {
   return NULL;
 }
 
+//=============================================================================
+//------------------------------Value------------------------------------------
+const Type* ConvHF2FNode::Value(PhaseGVN* phase) const {
+  const Type *t = phase->type( in(1) );
+  if( t == Type::TOP ) return Type::TOP;
+  if( t == TypeInt::SHORT ) return Type::FLOAT;
+  const TypeInt *ti = t->is_int();
+  if ( ti->is_con() ) return TypeF::make( SharedRuntime::hf2f( ti->get_con() ) );
+
+  return bottom_type();
+}
+
+//------------------------------Identity---------------------------------------
+Node* ConvHF2FNode::Identity(PhaseGVN* phase) {
+  return (in(1)->Opcode() == Op_ConvF2HF) ? in(1)->in(1) : this;
```

Comment:
We cannot do this optimization:
     HF2F(F2HF(x)) != x 
This is because precision and range both are lost when we convert F2HF.